src/utils.ts

```diff
@@ -187,10 +187,34 @@ export namespace Utils {
       );
     }
 
-    // verify homeDomain
-    if (homeDomain && `${homeDomain} auth` !== operation.name) {
+    // verify homeDomains
+    if (!homeDomains) {
+      throw new InvalidSep10ChallengeError(
+        "Invalid homeDomains: the transaction requires a home domain",
+      );
+    }
+
+    let matchedHomeDomain;
+
+    if (typeof homeDomains === "string") {
+      if (`${homeDomains} auth` === operation.name) {
+        matchedHomeDomain = homeDomains;
+      }
+    } else {
```

Comment:
I think we can make this logic a bit more readable if the `else` statement is instead a 
```
} else if (!Array.isArray(homeDomains)) {
    throw new InvalidSep10ChallengeError(
      `Invalid homeDomains: homeDomains type is ${typeof homeDomains} but should be a string or an array`,
    );
} else {
    matchedHomeDomain = homeDomains.find(
      (domain) => `${domain} auth` === operation.name,
    );
}
```

Reply:
or 
```
} else if (Array.isArray(homeDomains)) {
    matchedHomeDomain = homeDomains.find(
      (domain) => `${domain} auth` === operation.name,
    );
} else {
    throw new InvalidSep10ChallengeError(
      `Invalid homeDomains: homeDomains type is ${typeof homeDomains} but should be a string or an array`,
    );
}
```

Reply:
Good point. :+1: Will update.